lib/bamboo/adapters/sendgrid_adapter.ex

```diff
@@ -147,6 +148,17 @@ defmodule Bamboo.SendgridAdapter do
   defp put_text_body(body, %Email{text_body: nil}), do: body
   defp put_text_body(body, %Email{text_body: text_body}), do: Map.put(body, :text, text_body)
 
+  defp put_x_smtp_api(body, %Email{private: %{"x-smtpapi" => fields}} = email) do
+    # SendGrid will error with empty bodies, even while using templates.
+    # Sets a default `text_body` if one is not specified, allowing the consumer
+    # to negelect doing so themselves.
+    case {email.text_body, email.html_body} do
+      {nil, nil} -> put_text_body(body, %Email{email | text_body: " "})
+      _ -> body
+    end |> Map.put("x-smtpapi", Poison.encode!(fields))
+  end
```

Comment:
Is there any reason someone would set a body *and* use a Sendgrid template? If not then maybe we should raise if either of the bodies are set and then always set an empty text body. What do you think?